tests/legacy/unit-tests/rest-api/Tests/Version3/orders.php

```diff
@@ -682,6 +683,25 @@ public function test_update_order_add_coupons() {
 		$this->assertEquals( 200, $response->get_status() );
 		$this->assertCount( 1, $data['coupon_lines'] );
 		$this->assertEquals( '45.00', $data['total'] );
+
+		// Let's repeat, but this time we'll specify the item ID for the coupon: this is
+		// a readonly property and we expect the request to fail as a result.
+		$request = new WP_REST_Request( 'PUT', '/wc/v3/orders/' . $order->get_id() );
+		$request->set_body_params(
+			array(
+				'coupon_lines' => array(
+					array(
+						'id' => 123,
+						'code' => 'fake-coupon',
+					),
+				),
+			)
+		);
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertEquals( 'woocommerce_rest_coupon_item_id_readonly', $data['code'] );
```

Comment:
It felt logical to ad this scenario (a failing update involving coupons, where the item ID is specified for a `coupon_line` item) to this existing test but I'm definitely happy to break it out into its own test method if preferred.